### PR TITLE
Implemented: Added feature to integrate field mappings with API(#85zrhcq56)

### LINKED ISSUE #129: Integrate field mappings feature with API
## What is the motivation for adding/enhancing this feature?

Currently, we store field mappings in localstorage, use APIs to store it at backend.

APIs:

createDataManagerMapping
Parameters:
mappingPrefId (optional)
mappingPrefValue
mappingPrefTypeEnumId
fromDate
thruDate

updateDataManagerMapping
Parameters:
mappingPrefId (optional)
mappingPrefValue
mappingPrefTypeEnumId
fromDate
thruDate

deleteDataManagerMapping
Parameters:
mappingPrefId

## What are the acceptance criteria?

- [ ] ...

## Can you complete this feature request by yourself?

- [ ] YES
- [ ] NO

### PR DESCRIPTION
**Pull request by shashwatbangar**
### Related Issues

Closes #129 

### Screenshots of Visual Changes before/after (If There Are Any)

**IMPORTANT NOTICE** - Remember to add changelog entry

### Contribution and Currently Important Rules Acceptance

- [ ] I read and followed [contribution rules](https://github.com/hotwax/import#contribution-guideline)

### CODE REVIEW

File: src/components/UpdateMappingPopover.vue [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+  IonList,
+  IonListHeader,
+  alertController,
+  modalController,
+  popoverController
+} from "@ionic/vue";
+import { defineComponent } from "vue";
+import { copyOutline, flashOutline, pinOutline, timeOutline  } from 'ionicons/icons'
+import { mapGetters, useStore } from 'vuex'
+import JobHistoryModal from '@/components/JobHistoryModal.vue'
```
Comment (ymaheshwari1) [inline src/components/UpdateMappingPopover.vue:56]: JobHistoryModal is not available in components.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/components/MappingConfiguration.vue [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+  IonSelectOption,
+  modalController
+} from "@ionic/vue";
+import { defineComponent } from "vue";
+import { close, save, saveOutline } from "ionicons/icons";
+import { useStore, mapGetters } from "vuex";
+import { UserService } from "@/services/UserService";
+import { hasError } from '@/utils'
+import { DateTime } from 'luxon';
+import { showToast } from '@/utils';
```
Comment (ymaheshwari1) [inline src/components/MappingConfiguration.vue:93]: Use single import statements from same file.

File: src/views/FieldMappingModal.vue [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+      // so removed the characters for now
+      return text.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, '\\$&');
+    },
+    findTimeZone() { 
+      const regularExp = new RegExp(`${this.escapeRegExp(this.queryString)}`, 'i');
+      this.filteredTimeZones = this.timeZones.filter((timeZone: any) => {
+        return regularExp.test(timeZone.id) || regularExp.test(timeZone.label);
+      });
+    },
+    async getAvailableTimeZones() {
```
Comment (ymaheshwari1) [inline src/views/FieldMappingModal.vue:137]: I think there is no need to for the timeZone specific code in here.

File: src/views/FieldMappingModal.vue [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+    <!-- Empty state -->
+    <div class="empty-state" v-if="fieldMappings.length === 0">
+      <p>{{ $t("No fieldMapping found. Please create new.")}}</p>
+    </div>
+
+    <!-- Mappings -->
+    <div v-else>
+      <ion-list>
+        <ion-item>
+            <ion-label>{{ $t("Order ID") }}</ion-label>
```
Comment (ymaheshwari1) [inline src/views/FieldMappingModal.vue:28]: Improve indentation.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/components/MappingConfiguration.vue (hunk 2 above)
Comment (shashwatbangar) [reply src/components/MappingConfiguration.vue:71]: Done sir

File: src/views/FieldMappingModal.vue (hunk 3 above)
Comment (shashwatbangar) [reply src/views/FieldMappingModal.vue:136]: Sure sir.

File: src/views/FieldMappingModal.vue (hunk 4 above)
Comment (shashwatbangar) [reply src/views/FieldMappingModal.vue:28]: Done sir.

**Comment by ymaheshwari1** [comment]
Closing as will be handled in #159

**Pull request closed without merging**